### ISSUE TITLE
ACCEPT_EMPTY_STRING_AS_NULL_OBJECT not honored in xml module

### ISSUE THREAD

**Issue by mortenoh** [FasterXML/jackson-dataformat-xml#25]
I'm having some issues with the treatment of empty strings in the XML module.

Say I have this:

``` xml
class A {
 @JacksonXmlProperty( isAttribute = true )
 private String stringA;
 private String stringB;
}

A a = new A();
a.setStringA("");
a.setStringB("");
```

In JSON this would give me:

``` json
{ "stringA": "", "stringB": ""}
```

In XML this gives me:

``` xml
<A stringA="">
 <stringB />
</A>
```

Which means that the attribute is treated correctly (as according to
the defaults), but for the element, it shows up as null. And is
therefore deserialized back as a null string.

**Comment by cowtowncoder** [comment]
Since this is an old issue I am not sure if it can be reproduced as is. There are newer related issues as well, so I will close this one in favor of newer ones.

**Comment by cowtowncoder** [comment]
Turns out there was a unit test I had added earlier. Was able to fix but only for 3.0.0, unfortunately, not 2.9.
